runtime/redirector/redirector.c

```diff
@@ -463,6 +464,60 @@ parseMemorySizeValue(char *option)
 	return result;
 }
 
+#define GC_POLICY_OPTION "-Xgcpolicy:"
+#define LENGTH_GC_POLICY_OPTION (sizeof(GC_POLICY_OPTION) - 1)
+
+static void
+checkEnvOptions(char *envOptions, int *gcPolicy, char **xcompressedstr, char **xnocompressedstr, char **xjvmstr, int *xjvm, char **namedVM, size_t *nameLength, char **xmxstr)
+{
+	char *gcPolicyString = findStartOfMostRightOption(envOptions, GC_POLICY_OPTION);
+	if (NULL == gcPolicyString) {
+		if (hasEnvOption(envOptions, "-XX:+UseNoGC")) {
+			gcPolicyString = "nogc";
+		}
+	}
+	if (NULL != gcPolicyString) {
+		parseGCPolicy(gcPolicyString + LENGTH_GC_POLICY_OPTION, gcPolicy);
```

Comment:
I know this isn't new code but it looks like we index off the end of the buffer here if `gcPolicyString = "nogc"`.

Likely fixable by setting 
```
gcPolicyString = "-Xgcpolicy:nogc"
```

Reply:
Updated.